frontend/__tests__/app/enrollment/components/sse-connection-status.test.tsx

```diff
@@ -0,0 +1,34 @@
+import { render } from '@testing-library/react';
+import TaskSseConnectionStatus from '@/app/enrollment/components/sse-connection-status';
+import '@testing-library/jest-dom';
+
+let connected = true;
+
+jest.mock('@/lib/store/task-progress-store', () => ({
+    __esModule: true,
+    default: jest.fn(() => ({
+        eventSourceConnected: connected,
+    })),
+}));
+
+beforeEach(() => {
+    jest.clearAllMocks();
+    connected = true;
+});
+
+describe('TaskSseConnectionStatus', () => {
+  it('shows green indicator when connected', () => {
+    render(<TaskSseConnectionStatus />);
+    const indicator = document.querySelector('.bg-green-500');
+    expect(indicator).toBeInTheDocument();
+  });
+
+
+  it('shows red indicator when disconnected', () => {
+    connected = false;
+    render(<TaskSseConnectionStatus />);
+    const indicator = document.querySelector('.bg-red-500');
```

Comment:
If we want to target specific elements on the page we might want to add `data-test-id` attributes and use that. If there is any other element that applies `bg-red-500` (or green like the previous test) the test will incorrectly pass.